DEVELOPMENT.md

```diff
@@ -87,6 +87,27 @@ Maintain the same quality for production and test code.
 
 When writing a Git commit message, follow these [guidelines](https://chris.beams.io/posts/git-commit/).
 
+### Git merge strategy
+
+Pull requests are usually merged into `master` using the  [`rebase and merge`](https://docs.github.com/en/github/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#rebase-and-merge-your-pull-request-commits) strategy.
+
+A typical pull request should strive to just contain one commit - logically unrelated changes should generally be extracted
+into their own PRs.
+
+If a pull request does consist of multiple commits, it is expected that every prefix of it is correct. That is,
+there might be preparatory commits at the bottom of the stack that don't bring any value by themselves,
+but none of the commits should introduce an error that is fixed by some future commit.
```

Comment:
```suggestion
each of the commits should build successfully and pass the tests. For this reason the commit adding a feature or fixing a bug should also contain the tests instead of them being separate commits.
```

I feel this example makes it more obvious what the text is trying to say in my opinion.

Reply:
I think these two descriptions are similar but not equivalent. What if a feature is added, and then a later commit adds or improves the unit test suite? Are we enforcing that these should be the same commit? Adding or improving unit tests later seems fine from my POV but let me know what you think.

Reply:
Good point. The existing text is more precise and doesn't leave room for confusion. Let's keep as is.